pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -558,4 +546,134 @@ default boolean defaultEnableGroupTrim() {
     @Nullable
     WorkerManager getWorkerManager();
   }
+
+  /// A query that have been parsed, validates, transformed into a [RelNode] and optimized with Calcite.
+  ///
+  /// This represents the last point where Calcite is being used. This object can then be:
+  /// - Used to get the tables involved in the query (see [#getTableNames])
+  /// - Used to explain the query plan (see [#explain])
+  /// - Used to plan how to evaluate the query using Pinot Engine (see [#planQuery])
+  ///
+  /// Compiled queries are craeted by calling [QueryEnvironment#compile] and should be closed.
```

Comment:
Let's also document that closing the compiled query closes the passed in `PlannerContext`?

Reply:
I don't know if callers need to know about PlannerContext or whether the fact that CompiledQuery contains a PlannerContext is written in stone, so I preferred to be less specific. Anyway, I don't think we need to be super formal in these mostly inner APIs, so I'm going to add that